More windows polish

pkgw opened the pull request.

pkgw committed:
MNT: Re-rendered with conda-smithy 2.0.1

pkgw committed:
Improve package description in meta.yaml.

pkgw committed:
Yet another pass at improving the Windows build process

The main change is to run autoreconf and use $LIBRARY_PREFIX. The former isn't
really necessary here but I want things to be uniform.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

pkgw committed:
Missed an aclocal include directory.

pkgw committed:
And yet more Windows details to work out.

$PKG_CONFIG_PATH needs $LIBRARY_PREFIX in Unixy form so I guess we just need
to bite the bullet and provide both versions. And I had a patch to 'configure'
that needs to become a patch to 'configure.ac' since we're running autoreconf
now.

pkgw committed:
Final $LIBRARY_PREFIX related futzing, I hope.
